feat(layout): Add `Flex::SpaceEvenly` (#1952)
ratatui/ratatui: BREAKING-CHANGES.md

Author: Done!

This is a quick summary of the sections below:

- [v0.30.0 Unreleased](#v0300-unreleased)

>   - Old `Flex::SpaceAround` behavior is available by using `Flex::SpaceEvenly` and new
>     `Flex::SpaceAround` now distributes space evenly around each element except the middle spacers

Reviewer: I find it a bit too long for summary, it's better to suggest a solution in details below. I think it can be just:
```suggestion
  - `Flex::SpaceAround` now mirrors flexbox: space between items is twice the size of the outer gaps
```

## v0.30.0 Unreleased

> ### Old `Flex::SpaceAround` behavior is available by using `Flex::SpaceEvenly` and new `Flex::SpaceAround` now distributes space evenly around each element except the middle spacers are twice the size of first and last elements ([#1952])

Reviewer: Same as in summary:
```suggestion
### `Flex::SpaceAround` now mirrors flexbox: space between items is twice the size of the outer gaps ([#1952])
```

[#1952]: https://github.com/ratatui/ratatui/pull/1952

The old `Flex::SpaceAround` behavior has been changed to distribute space evenly around each

> element, with the middle spacers being twice the size of the first and last elements. The old

Reviewer: ```suggestion
element, with the middle spacers being twice the size of the first and last one. The old
```
